qiskit/aqua/algorithms/phase_estimators/phase_estimator_result.py

```diff
@@ -0,0 +1,180 @@
+        of the unitary so that the peak of the probability density occurs at the bit string
+        that most closely approximates the true phase.
+        """
+        if self._phase_dict is not None:
+            binary_phase_string = max(self._phase_dict, key=self._phase_dict.get)
+        else:
+            # numpy.argmax ignores complex part of number. But, we take abs anyway
+            idx = numpy.argmax(abs(self._phase_array))
+            binary_phase_string = numpy.binary_repr(idx, self._num_evaluation_qubits)[::-1]
+        phase = _bit_string_to_phase(binary_phase_string)
+        return phase
+
+    def filter_phases(self, cutoff: float = 0.0,
+                      as_float: bool = True) -> Dict:
+        """Return a filtered dict of phases (keys) and frequencies (values).
+
+        Only phases with frequencies (counts) larger than `cutoff` are included.
+        It is assumed that the `run` method has been called so that the phases have been computed.
+        When using a noiseless, shot-based simulator to read a single phase that can
+        be represented exactly by `num_evaluation_qubits`, all the weight will
+        be concentrated on a single phase. In all other cases, many, or all, bit
+        strings will have non-zero weight. This method is useful for filtering
+        out these uninteresting bit strings.
+
+        Args:
+            cutoff: Minimum weight of number of counts required to keep a bit string.
+                    The default value is `0.0`.
+            as_float: If `True`, returned keys are floats in `[0.0, 1)`. If `False`
+                      returned keys are bit strings.
+
+        Returns:
+            A filtered dict of phases (keys) and frequencies (values).
+        """
+        if self._phase_dict is not None:
+            counts = self._phase_dict
+            if as_float:
+                phases = {_bit_string_to_phase(k): counts[k]
+                          for k in counts.keys() if counts[k] > cutoff}
+            else:
+                phases = {k: counts[k] for k in counts.keys() if counts[k] > cutoff}
+
+        else:
+            phases = {}
+            for idx, amplitude in enumerate(self._phase_array):
+                if amplitude > cutoff:
+                    # Each index corresponds to a computational basis state with the LSB rightmost.
+                    # But, we chose to apply the unitaries such that the phase is recorded
+                    # in reverse order. So, we reverse the bitstrings here.
+                    binary_phase_string = numpy.binary_repr(idx, self._num_evaluation_qubits)[::-1]
+                    if as_float:
+                        _key = _bit_string_to_phase(binary_phase_string)
+                    else:
+                        _key = binary_phase_string
+                    phases[_key] = amplitude
+
+            phases = _sort_phases(phases)
+
+        return phases
+
+
+# TODO: maybe we do want to remove leading _
```

Comment:
Is this TODO still open? I think this should be a private function, this is such minimal logic that it is probably easier for the user to just implement themselves instead of importing it from Qiskit (and looking up what exactly it does 🙂 )

Reply:
I'll close it for now. I'd have to have a compelling use in a tutorial or something, which I don't have yet. I'm not convinced it's easier for a user to implement this themselves, especially if they are not very familiar with python and/or QPE. Suppose a user is trying to understand and debug at the same time. Do you need to reverse the bitstring before converting ? It should be documented better. But, the idea is that:
you pass the function a bitstring that was returned by this algorithm, and it returns `phi`.
You don't need to understand any details; reverse or not ? Did I miss a factor of 2 ?
That's the motivation anyway. You could argue that laying it out in a tutorial is better than putting it in the API.